src/Bicep.Core/DataFlow/LocalSymbolDependencyVisitor.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Bicep.Core.DataFlow
+{
+    public sealed class LocalSymbolDependencyVisitor : SyntaxVisitor
+    {
+        private readonly SemanticModel semanticModel;
+
+        private LocalSymbolDependencyVisitor(SemanticModel semanticModel)
+        {
+            this.semanticModel = semanticModel;
+        }
+
+        private HashSet<LocalVariableSymbol> SymbolDependencies { get; } = new();
+
+        public static ISet<LocalVariableSymbol> GetLocalSymbolDependencies(SemanticModel semanticModel, SyntaxBase syntax)
```

Comment:
Worth making immutable?

Reply:
Yeah good pt.

---
In reply to: [582071733](https://github.com/Azure/bicep/pull/1521#discussion_r582071733) [](ancestors = 582071733)

Reply:
Done.